CMakeLists.txt

```diff
@@ -67,13 +67,35 @@ function(trusted_firmware_build)
   set(PSA_API_NS_PATH ${TFM_BINARY_DIR}/interface/libpsa_api_ns.a)
   set(TFM_GENERATED_INCLUDES ${TFM_BINARY_DIR}/generated/interface/include)
 
-  set(BL2_HEX_FILE ${CMAKE_BINARY_DIR}/tfm/bin/bl2.hex)
+  set(BL2_BIN_PATH ${TFM_BINARY_DIR}/bin/bl2.bin)
+  set(BL2_HEX_PATH ${TFM_BINARY_DIR}/bin/bl2.hex)
+  set(BL2_ELF_PATH ${TFM_BINARY_DIR}/bin/bl2.elf)
+  set(TFM_S_BIN_PATH ${TFM_BINARY_DIR}/bin/tfm_s.bin)
+  set(TFM_S_HEX_PATH ${TFM_BINARY_DIR}/bin/tfm_s.hex)
+  set(TFM_S_ELF_PATH ${TFM_BINARY_DIR}/bin/tfm_s.elf)
+  set(TFM_NS_BIN_PATH ${TFM_BINARY_DIR}/bin/tfm_ns.bin)
+  set(TFM_NS_HEX_PATH ${TFM_BINARY_DIR}/bin/tfm_ns.hex)
+  set(TFM_NS_ELF_PATH ${TFM_BINARY_DIR}/bin/tfm_ns.elf)
+  set(TFM_S_SIGNED_BIN_PATH ${TFM_BINARY_DIR}/bin/tfm_s_signed.bin)
+  set(TFM_NS_SIGNED_BIN_PATH ${TFM_BINARY_DIR}/bin/tfm_ns_signed.bin)
+  set(TFM_S_NS_SIGNED_BIN_PATH ${TFM_BINARY_DIR}/bin/tfm_s_ns_signed.bin)
```

Comment:
You're right, TFM creates axf, and elf files are explicitly generated from that. To be clear, should I rename _ELF_FILE to _FILE? I'm not in favor myself (more ambiguous for less experieced Cmake users).

Reply:
I think names are clear for `_BIN_` and `_HEX_` files, as they are not regular CMake output files.

For the elf / axf file, i'm a little in doubt.

For which purpose are those included ?
Bin and hex are clear, cause they will be used for embedding into another image, but the elf / axf is not that obvious.

If they are included because the are useful for debugging or direct flashing, then both of them are good and I would suggest to follow the CMake style where the build system decides how `TARGET_FILE` looks.
In which case I believe the axf file might be more correct.

But I need a little more background knowledge to know if/how the axf/elf file is intended to be used.

Reply:
The elf files might not be needed yet. Should I remove them?

Reply:
then let's keep it simple.
very easy to extend if need be.

Reply:
Ok, removing the elf files

Reply:
Done